src/elements/Label/Label.js

```diff
@@ -31,31 +32,30 @@ function Label(props) {
     size,
     color,
     useKeyOnly(basic, 'basic'),
+    useKeyOnly(circular, 'circular'),
     useKeyOnly(floating, 'floating'),
     useKeyOnly(horizontal, 'horizontal'),
+    useKeyOnly(empty, 'empty'),
     useKeyOnly(tag, 'tag'),
     useValueAndKey(attached, 'attached'),
     useKeyOrValueAndKey(corner, 'corner'),
     useKeyOrValueAndKey(pointing, 'pointing'),
     useKeyOrValueAndKey(ribbon, 'ribbon'),
-    circular && (children && 'circular' || 'empty circular'),
     // TODO how to handle image child with no image class? there are two image style labels.
     (image || childrenUtils.someByType(children, Image) || childrenUtils.someByType(children, 'img')) && 'image',
     'label',
     className
   )
 
-  const DetailComponent = (detailLink || onDetailClick) && 'a' || 'div'
-  const ElementType = getElementType(Label, props, () => {
-    if (image || link || onClick) return 'a'
-  })
+  const DetailComponent = detailAs || 'div'
```

Comment:
Hm.  I think here we should be using `getElementType()` for consistent handling.  This makes me realize that this component is actually missing sub components.  It should have `Label.Detail` for instance.  That would also solve all the issues related to it.

If you like, we can merge this as is, and open an issue to create Label sub components.